ibctest/relayer.go

```diff
@@ -0,0 +1,256 @@
+package ibctest
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/cosmos/cosmos-sdk/crypto/keyring"
+	"github.com/cosmos/relayer/v2/cmd"
+	"github.com/cosmos/relayer/v2/internal/relayertest"
+	"github.com/cosmos/relayer/v2/relayer/provider/cosmos"
+	"github.com/ory/dockertest/v3"
+	"github.com/strangelove-ventures/ibctest/ibc"
+	"go.uber.org/zap"
+	"go.uber.org/zap/zaptest"
+)
+
+type Relayer struct {
+	t *testing.T
+
+	pool *dockertest.Pool
+
+	home string
+
+	// Set during StartRelayer.
+	errCh  chan error
+	cancel context.CancelFunc
+}
+
+func (r *Relayer) sys() *relayertest.System {
+	return &relayertest.System{HomeDir: r.home}
+}
+
+func (r *Relayer) log() *zap.Logger {
+	return zaptest.NewLogger(r.t)
+}
+
+func (r *Relayer) AddChainConfiguration(ctx context.Context, _ ibc.RelayerExecReporter, chainConfig ibc.ChainConfig, keyName, rpcAddr, grpcAddr string) error {
+	sys := &relayertest.System{HomeDir: r.home}
+	sys.MustAddChain(r.t, chainConfig.ChainID, cmd.ProviderConfigWrapper{
+		Type: "cosmos",
+		Value: cosmos.CosmosProviderConfig{
+			Key:     keyName,
+			ChainID: chainConfig.ChainID,
+			RPCAddr: rpcAddr,
+			// GRPCAddr: grpcAddr, // Not part of relayer cosmos provider config (yet)
+			AccountPrefix:  chainConfig.Bech32Prefix,
+			KeyringBackend: keyring.BackendTest,
+			GasAdjustment:  chainConfig.GasAdjustment,
+			GasPrices:      chainConfig.GasPrices,
+			Debug:          true,
+			Timeout:        "10s",
+			OutputFormat:   "json",
+			SignModeStr:    "direct",
+		},
+	})
+
+	return nil
+}
+
+func (r *Relayer) AddKey(ctx context.Context, _ ibc.RelayerExecReporter, chainID, keyName string) (ibc.RelayerWallet, error) {
+	res := r.sys().RunC(ctx, r.log(), "keys", "add", chainID, keyName)
+	if res.Err != nil {
+		return ibc.RelayerWallet{}, res.Err
+	}
+
+	var w ibc.RelayerWallet
+	if err := json.Unmarshal(res.Stdout.Bytes(), &w); err != nil {
+		return ibc.RelayerWallet{}, err
+	}
+
+	return w, nil
+}
+
+func (r *Relayer) RestoreKey(ctx context.Context, _ ibc.RelayerExecReporter, chainID, keyName, mnemonic string) error {
+	res := r.sys().RunC(ctx, r.log(), "keys", "restore", chainID, keyName, mnemonic)
+	if res.Err != nil {
+		return res.Err
+	}
+	return nil
+}
+
+func (r *Relayer) GeneratePath(ctx context.Context, _ ibc.RelayerExecReporter, srcChainID, dstChainID, pathName string) error {
+	res := r.sys().RunC(ctx, r.log(), "paths", "new", srcChainID, dstChainID, pathName)
+	if res.Err != nil {
+		return res.Err
+	}
+	return nil
+}
+
+func (r *Relayer) ClearQueue(ctx context.Context, _ ibc.RelayerExecReporter, pathName string, channelID string) error {
+	panic("not yet implemented")
+}
```

Comment:
`ClearQueue` is no longer part of the ibctest Relayer interface, so this method should be able to be deleted.

Reply:
removed